mqtt/mqtt-broker/tests/smoke.rs

```diff
@@ -113,3 +115,153 @@ async fn basic_pub_sub() {
         .unwrap()
         .expect("can't wait for the broker");
 }
+
+#[tokio::test]
+async fn retained_messages() {
```

Comment:
can we make test case names with more context in it? not just it relates to the retained messages

Reply:
This comment applicable to all test cases

Reply:
As discussed offline: we'll add test comments that explain the scenario instead.